Developers/Unity Project/Spooksville/Assets/Scripts/BattleManager.cs

```diff
@@ -1,29 +1,81 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.UI;
 
 public class BattleManager : MonoBehaviour
 {
     public static BattleManager instance;
 
-    private int bossHealth = 100;
+    [Header("Containers")]
+    [SerializeField] private Text headerContainer;
+    private Text previousHeaderText;
+    public Text centerInventoryContainer;
+    public List<Text> inventoryContainers;
 
-    void Start()
+    [Header("Boss Settings")]
+    public int bossHealth = 100;
+
+    private string weaponSelectionMessage = "What weapon will you choose?";
+
+    private void Start()
     {
         if (instance == null)
+        {
             instance = this;
+        }
+
+        Inventory.Hide();
+    }
+
+    private void Update()
+    {
+        Inventory.InventorySelection();
     }
 
+    #region Fight Logic
     public void Attack(Weapon weapon)
     {
-        bossHealth -= weapon.damage;
+        bossHealth -= weapon.Damage;
+
+        DisplayAttackText("Mom was attacked by " + weapon.Name);
+    }
+    #endregion
+
+    #region UI Management
+    public void DisplayHeaderText(string text)
+    {
+        previousHeaderText.text = headerContainer.text;
 
-        //Delay when showing the text before resetting UI
-        Invoke("ResetUI", 3);
+        headerContainer.text = text;
     }
 
-    private void ResetUI()
+    public void DisplayTemporaryHeaderText(string text, float time)
     {
+        headerContainer.text = text;
+        StartCoroutine(WaitSeconds(time));
+        PreviousHeaderText();
+    }
+
+    private void DisplayAttackText(string text)
+    {
+        headerContainer.text = text;
+
+        Inventory.Hide();
+        StartCoroutine(WaitSeconds(3f));
+
+        headerContainer.text = previousHeaderText.text;
```

Comment:
FYI this is going to fire immediately. If you want it to happen after 3 seconds it needs to be inside a coroutine, after the yield statement.